file-formats/cfdf/type/zif_aff_cfdf_v1.intf.abap

```diff
@@ -0,0 +1,592 @@
+      characteristic_name TYPE c LENGTH 30,
+
+      "! <p class="shorttext">Class Name</p>
+      "! Class name
+      "! $maxLength 3
+      class_type          TYPE c LENGTH 3,
+
+      "! <p class="shorttext">Object Type</p>
+      "! Object type
+      "! $maxLength 30
+      object_type         TYPE c LENGTH 30,
+    END OF ty_gs_characteristic_reference.
+
+
+  TYPES:
+    "! <p class="shorttext">Custom Field</p>
+    "! Custom field
+    BEGIN OF ty_main,
+      "! $required
+      format_version              TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header                      TYPE zif_aff_types_v1=>ty_header_60,
+
+      "! <p class="shorttext">External Field Name</p>
+      "! External field name
+      "! $required
+      "! $maxLength 30
+      external_field_name         TYPE ty_field_name,
+
+      "! <p class="shorttext">Type</p>
+      "! Custom field type
+      "! $required
+      BEGIN OF type,
+        "! <p class="shorttext">Semantic Field Type</p>
+        "! Semantic field type
+        "! $required
+        "! $maxLength 10
+        "! $values {@link zif_aff_cfdf_v1.data:gc_type}
+        semantic_field_type        TYPE ty_field_type,
+
+        "! <p class="shorttext">Length</p>
+        "! Length
+        "! $maximum 500
+        length                     TYPE i,
+
+        "! <p class="shorttext">Scale</p>
+        "! Scale
+        "! $maximum 14
+        scale                      TYPE i,
+
+        "! <p class="shorttext">Aggregation</p>
+        "! Aggregation
+        "! $maxLength 5
+        "! $values {@link zif_aff_cfdf_v1.data:gc_aggregation}
+        "! $default {@link zif_aff_cfdf_v1.data:gc_aggregation.none}
+        aggregation                TYPE ty_aggregation,
+
+        "! <p class="shorttext">Code List Uppercase Only</p>
+        "! Code List uppercase only
+        code_list_uppercase_only   TYPE abap_bool,
+
+        "! <p class="shorttext">Associated Business Object</p>
+        "! Associated business object
+        "! $maxLength 30
+        associated_business_object TYPE c LENGTH 30,
+
+        "! <p class="shorttext">Value Help View</p>
+        "! Value help view
+        "! $maxLength 30
+        value_help_view            TYPE ty_cds_view_name,
+
+        "! <p class="shorttext">Dimension View</p>
+        "! Dimension view
+        "! $maxLength 30
+        dimension_view             TYPE ty_cds_view_name,
+      END OF type,
+
+      "! <p class="shorttext">Data Protection and Privacy</p>
+      "! Data protection and privacy
+      data_protection_and_privacy TYPE ty_gs_dpp_properties,
+
+      "! <p class="shorttext">Reference to Characteristic</p>
+      "! Reference to characteristic
+      characteristic_reference    TYPE ty_gs_characteristic_reference,
+
+      "! <p class="shorttext">Label and Tooltip</p>
+      "! Label and tooltip
+      ui_texts                    TYPE ty_gt_ui_text,
```

Comment:
Differences between title and field name. If you want to stick to "Label and Tooltip" I would change it to plural, because these seem to be multiple values

Reply:
Title has been adapted to plural, however we would like to stick to the difference between title and field name, if possible

Reply:
Sure, this is technically possible. 

I have just added a comment, because usually this is an indicator that one of both should be renamed ;)